sway-lib-std/src/primitive_conversions/str.sw

```diff
@@ -6,17 +6,23 @@ impl str {
     pub fn try_as_str_array<S>(self) -> Option<S> {
         __assert_is_str_array::<S>();
         let str_size = __size_of_str_array::<S>();
+        let tmp_alloc_size = __size_of::<S>();
         let source = self.as_ptr();
 
         if self.len() == str_size {
-            let s: S = asm(str_size: str_size, source: source, dest) {
+            let s: S = asm(
+                str_size: str_size,
+                tmp_alloc_size: tmp_alloc_size,
+                source: source,
+                dest,
+            ) {
                 move dest sp;
-                cfe str_size;
+                cfe tmp_alloc_size;
```

Comment:
Have we analyzed other places where we allocate/modify stack/heap? Maybe we have this problem in other places as well.

Reply:
There's a similar code in `from_str_array` in `core::str`, but that already is doing it right, by using `__size_of` for the allocation.

I checked other uses of `cfe` and `alloc` in `std` and `core`, and they all seem fine to me.